### PR TITLE
jinjalint checks - half of them!

### PR DESCRIPTION
**Pull request by minusworld**
Adding documentation for the following checks:

- check_csrf_protection
- check_anchor_target_blank_noopener
- check_anchor_target_blank_noreferrer
- check_anchor_href_template

### CODE REVIEW

File: docs/jinjalint/anchor-missing-noopener.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# jinjalint-anchor-missing-noopener
+
+**tl;dr** Pages opened with `target="_blank"` allow the new page to access the original's `window.opener`. This can have security, privacy, and performance implications. Include `rel="noopner noreferrer"` to prevent this.
```
Comment (mschwager) [inline docs/jinjalint/anchor-missing-noopener.md]: The `noopener` attribute value impacts the security and performance while `noreferrer` impacts security, performance, and privacy (`noreferrer` implies `noopener`). We may want to split these two descriptions here and below to only include the relevant concerns. That way if someone doesn't care about `noreferrer` they can simply ignore that rule, and vice versa.

Also, looks like there's a typo here: `noopner` -> `noopener`.

File: docs/jinjalint/anchor-missing-noopener.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Description
+
+[This page](https://developers.google.com/web/tools/lighthouse/audits/noopener) by Google explains the danger of `target="_blank"`. The short version is that a page opened with `target="_blank"` can access the `window` object of the origin page. It can also manipulate the `window.opener` property, which could redirect the origin page to a malicious URL. This is called [**reverse tabnapping**](https://owasp.org/www-community/attacks/Reverse_Tabnabbing).
```
Comment (mschwager) [inline docs/jinjalint/anchor-missing-noopener.md]: How does this sound:

"[Google Lighthouse](<link>) recommends including `noopener` and `noreferrer` when using `target="_blank"`. In short, a page opened with ..."

File: docs/jinjalint/anchor-missing-noreferrer.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# jinjalint-anchor-missing-noreferrer
+
+**tl;dr** Pages opened with `target="_blank"` allow the new page to access the original's `window.opener`. This can have security, privacy, and performance implications. Include `rel="noopner noreferrer"` to prevent this.
```
Comment (mschwager) [inline docs/jinjalint/anchor-missing-noreferrer.md]: See above - and typo with `noopner` -> `noopener`.

File: docs/jinjalint/form-missing-csrf-protection.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# jinjalint-form-missing-csrf-protection
+
+**tl;dr** Flask apps using Flask-WTF require including a CSRF token in the HTML template itself. This check detects missing CSRF protection in HTML forms in Jinja templates.
+
+## Description
+
+The [Flask-WTF documentation](https://flask-wtf.readthedocs.io/en/stable/csrf.html) states that forms must render the CSRF token in the template. It is highly recommended that all forms are protected with CSRF tokens.
```
Comment (mschwager) [inline docs/jinjalint/form-missing-csrf-protection.md]: I think we can drop the initial "The" here.

File: docs/jinjalint/anchor-missing-noopener.md (hunk 1 above)
Comment (minusworld) [reply docs/jinjalint/anchor-missing-noopener.md]: Ohhh, yeah. I didn't fully understand that. Thanks! 👍

File: docs/jinjalint/anchor-missing-noopener.md (hunk 2 above)
Comment (minusworld) [reply docs/jinjalint/anchor-missing-noopener.md]: That's much better. I was trying to figure out how to word it better. Thanks!

File: docs/jinjalint/form-missing-csrf-protection.md (hunk 4 above)
Comment (minusworld) [reply docs/jinjalint/form-missing-csrf-protection.md]: 👍

**Comment by minusworld** [comment]
OK, I've addressed the comments. Thanks! 👍

**Pull request merged**